site/content/docs/user/ingress.md

```diff
@@ -0,0 +1,168 @@
+---
+title: "Ingress"
+menu:
+  main:
+    parent: "user"
+    identifier: "user-ingress"
+    weight: 3
+---
+
+## Setting Up An Ingress Controller
+
+We can leverage KIND's `extraPortMapping` config option when creating a cluster to
+forward ports from the host to an ingress controller running on a node. We can also specify 
+custom node label by using `node-labels` in the kubeadm `InitConfiguration`, to be used
+by the ingress controller `nodeSelector`.
+The following ingress controllers are known to work:
+
+ - [Ingress NGINX](#ingress-nginx)
+
+### Ingress NGINX
+
+The following shell script will create a kind cluster, deploy 
+the standard ingress-nginx components, and apply the necessary patches.
+
+```bash 
+#!/bin/sh
+set -o errexit
+
+# Create cluster with hostPorts opened for http and https
+cat <<EOF | kind create cluster --config=-
```

Comment:
let's put all of these as files + inline their contents
maybe kind should support config files over HTTPS :thinking: